src/ILCompiler.TypeSystem/tests/GenericTypeAndMethodTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Internal.TypeSystem;
+using Xunit;
+
+namespace TypeSystemTests
+{
+    public class GenericMethodTests
+    {
+        private TestTypeSystemContext _context;
+        private ModuleDesc _testModule;
+
+        public GenericMethodTests()
+        {
+            _context = new TestTypeSystemContext(TargetArchitecture.Unknown);
+            var systemModule = _context.CreateModuleForSimpleName("CoreTestAssembly");
+            _context.SetSystemModule(systemModule);
+
+            _testModule = systemModule;
+        }
+
+        /// <summary>
+        /// Testing proper instantiation of methods involving generic parameters in their signature.
+        /// </summary>
+        [Fact]
+        [ActiveIssue(-1)]
+        public void TestMethodInstantiation()
+        {
+            MetadataType t = _testModule.GetType("GenericTypes", "GenericClass`1");
+            MethodDesc fooMethod = t.GetMethods().First(m => m.Name == "Foo");
+            MethodDesc instantiatedMethod =
+                fooMethod.InstantiateSignature(
+                    new Instantiation(new TypeDesc[1] {_context.GetWellKnownType(WellKnownType.Int32)}),
+                    Instantiation.Empty);
+
+            Assert.True(instantiatedMethod.Signature.ReturnType == _context.GetWellKnownType(WellKnownType.Int32));
+        }
+
+        /// <summary>
+        /// Testing lookup up of a method in an instantiated type.
+        /// </summary>
+        [Fact]
+        [ActiveIssue(-1)]
+        public void TestMethodLookup()
+        {
+            MetadataType t = _testModule.GetType("GenericTypes", "NonGenericClass");
+            MethodDesc barMethod = t.GetMethods().First(m => m.Name == "Bar");
+            // Quickly get a properly instantiated generic type GenericClass<Int32>
```

Comment:
```
_testModule.GetType("GenericTypes", "GenericClass`1").MakeInstantiatedType(new Instantiation(new TypeDesc[] { _context.GetWellKnownType(WellKnownType.Int32) }));
```

This is shorter than all the comments trying to explain what this does (here, and in GenericTypes.cs). If you have to spend more space commenting on what a thing does vs. just rewriting it, I would vote for rewriting it.

Feel free to add an overload to `MakeInstantiatedType` that takes `params TypeDesc[]` to make this shorter (it's general goodness and I wanted to do it for a while).